# HG changeset patch
# User Marco Caliari marco.caliari@univr.it
# Date 1304566450 25200
#      Wed May 04 20:34:10 2011 -0700
# Branch stable
# Node ID c6378cec77af0b406ccf9a3c114f64022957ab18
# Parent  e38fb19105636e281ea2e0125bc55e69f441d46b
Fix unfilled contourf when lvl = [0,0] (bug #33177).

* __contour__.m: Add sqrt(eps) to lvl_eps to prevent it from ever
being exactly zero.

diff --git a/scripts/plot/private/__contour__.m b/scripts/plot/private/__contour__.m
--- a/scripts/plot/private/__contour__.m
+++ b/scripts/plot/private/__contour__.m
@@ -525,17 +525,17 @@ function update_text (h, d)
   endif
 endfunction
 
 function lvl_eps = get_lvl_eps (lev)
   ## FIXME -- is this the right thing to do for this tolerance?  Should
   ## it be an absolute or relative tolerance, or switch from one to the
   ## other depending on the value of lev?
   if (isscalar (lev))
-    lvl_eps = abs (lev) * sqrt (eps);
+    lvl_eps = abs (lev) * sqrt (eps) + sqrt (eps);
   else
     tmp = min (abs (diff (lev)));
     if (tmp < 10*eps)
       lvl_eps = sqrt (eps);
     else
       lvl_eps = tmp / 1000.0;
     endif
   endif
